python/TestHarness/testers/RunApp.py

```diff
@@ -254,16 +254,16 @@ def testExitCodes(self, moose_dir, options, output):
             elif self.exit_code != 0 and specs['should_crash'] == False:
                 # Let's look at the error code to see if we can perhaps further split this out later with a post exam
                 reason = 'CRASH'
-                return "\n\nExit Code: " + str(self.exit_code)
             # Valgrind runs
             elif self.exit_code == 0 and self.shouldExecute() and options.valgrind_mode != '' and 'ERROR SUMMARY: 0 errors' not in output:
                 reason = 'MEMORY ERROR'
 
             if reason != '':
-                self.setStatus(self.fail, reason)
+                self.setStatus(self.fail, str(reason))
+                reason += "\n\nExit Code: " + str(self.exit_code)
```

Comment:
Hmm - are you sure you want to overload the meaning of "reason" here? I can see that this logic is OK because you are setting the status then changing the value of reason. This is not a very good design because if something gets reordered (think about what would happen if the setStatus method got moved around), the reason variable might contain the "reason" AND some extra newlines and an exit code. How about creating a second local variable that you use to hang onto any extra stuff that you want to print out? Or you could just return that stuff directly.